src/components/PaginatedTable/TableChunk.tsx

```diff
@@ -1,34 +1,109 @@
 import React from 'react';
 
+import {tableDataApi} from '../../store/reducers/tableData';
+import type {IResponseError} from '../../types/api/error';
 import {getArray} from '../../utils';
+import {useDelayed} from '../../utils/hooks/useDelayed';
+import {ResponseError} from '../Errors/ResponseError';
 
-import {LoadingTableRow, TableRow} from './TableRow';
-import type {Chunk, Column, GetRowClassName} from './types';
+import {EmptyTableRow, LoadingTableRow, TableRow} from './TableRow';
+import type {Column, FetchData, GetRowClassName, PaginatedTableData, SortParams} from './types';
 
-// With original memo generic types are lost
-const typedMemo: <T>(Component: T) => T = React.memo;
+const DEBOUNCE_TIMEOUT = 200;
 
-interface TableChunkProps<T> {
+interface TableChunkDataLoaderProps<T, F> {
+    id: number;
+    chunkSize: number;
+    sortParams?: SortParams;
+    filters?: F;
+
+    fetchData: FetchData<T, F>;
+    onDataStateChange: (data?: PaginatedTableData<T>) => void;
+    onErrorStateChange: (error?: unknown) => void;
+    onLoadingStateChange: (isLoading: boolean) => void;
+}
+
+const TableChunkDataLoader = <T, F>({
```

Comment:
This component seems to overcomplicate logic - it requires a few `setState` and `useEffects` just to duplicate `useFetchTableChunkQuery` state. Lets move this logic to `TableChunk`?